Entities/AtomGroup.cpp

```diff
@@ -1,2618 +1,1750 @@
-        massB = pIntersectedMO->GetMass();
-        invMassA = 1.0F / massA;
-        invMassB = 1.0F / massB;
-        // Normalize the masses
-        normMassA = invMassA / (invMassA + invMassB);
-        normMassB = invMassB / (invMassA + invMassB);
-
-        // If the intersected is much larger than this' MO, then only move this
-        if (normMassB < 0.33F)
-            thisExit = totalExitVector;
-        // Apply the movements to both this and the intersected MO's, proportional to their respective masses
-        else
-        {
-            thisExit = totalExitVector * normMassA;
-            intersectedExit = -totalExitVector * normMassB;
-        }
-    }
-
-    // Now actually apply the exit vectors to both, but only if the jump isn't too jarring
-    if (thisExit.GetMagnitude() < m_pOwnerMO->GetRadius())
-        position += thisExit;
-    if (!intersectedExit.IsZero() && intersectedExit.GetMagnitude() < pIntersectedMO->GetRadius())
-        pIntersectedMO->SetPos(pIntersectedMO->GetPos() + intersectedExit);
-
-    // If we've been pushed into the terrain enough, just gib as the squashing effect
-    if (m_pOwnerMO->CanBeSquished() && RatioInTerrain() > 0.75F)// && totalExitVector.GetMagnitude() > m_pOwnerMO->GetDiameter())
-    {
-        // Move back before gibbing so gibs don't end up inside terrain
-        position -= thisExit;
-        m_pOwnerMO->GibThis(-totalExitVector);
-    }
-
-    pIntersectedMOS = dynamic_cast<MOSRotating *>(pIntersectedMO);
-
-    if (pIntersectedMOS && pIntersectedMOS->CanBeSquished() && pIntersectedMOS->GetAtomGroup()->RatioInTerrain() > 0.75F)// && totalExitVector.GetMagnitude() > pIntersectedMO->GetDiameter())
-    {
-        // Move back before gibbing so gibs don't end up inside terrain
-        pIntersectedMO->SetPos(pIntersectedMO->GetPos() - intersectedExit);
-        pIntersectedMOS->GibThis(totalExitVector);
-    }
-
-// TODO: this isn't really true since we don't check for clearness after moving the position
-    return intersectingAtoms.empty();
-}
-
-
-void AtomGroup::AddAtomToGroup(int x, int y, const Vector &spriteOffset, MOSRotating *pOwnerMOSRotating, bool calcNormal)
-{
-	Atom *pAtom = new Atom;
-	pAtom->Create(Vector(static_cast<float>(x) + spriteOffset.m_X, static_cast<float>(y) + spriteOffset.m_Y), m_pMaterial, pOwnerMOSRotating);
-	if (calcNormal) {
-		pAtom->CalculateNormal(pOwnerMOSRotating->GetSpriteFrame(), -(pOwnerMOSRotating->GetSpriteOffset()));
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void AtomGroup::AddAtomToGroup(MOSRotating *ownerMOSRotating, const Vector &spriteOffset, int x, int y, bool calcNormal) {
+		Atom *atomToAdd = new Atom(Vector(static_cast<float>(x) + spriteOffset.m_X, static_cast<float>(y) + spriteOffset.m_Y), m_Material, ownerMOSRotating);
+		if (calcNormal) { atomToAdd->CalculateNormal(ownerMOSRotating->GetSpriteFrame(), -(ownerMOSRotating->GetSpriteOffset())); }
```

Comment:
I don't think the brackets in `-(ownerMOSRotating->GetSpriteOffset())` help readability or anything here, having the `-` without brackets is fine imo.